docs/Extended_dry_run.rst

```diff
@@ -0,0 +1,731 @@
+* :ref:`extended_dry_run_guidelines_easyblocks_framework_functions_runtime_patching`
+* :ref:`extended_dry_run_guidelines_easyblocks_framework_functions_run_cmd`
+
+.. _extended_dry_run_guidelines_easyblocks_framework_functions_setvar:
+
+Defining environment variables: ``setvar``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For defining environment variables, the ``setvar`` function available in the ``easybuild.tools.environment`` module
+should be used.
+
+For example, from the WRF easyblock::
+
+    jasper = get_software_root('JasPer')
+    if jasper:
+        env.setvar('JASPERINC', os.path.join(jasper, 'include'))
+
+When triggered in dry run mode, this will result in a clear dry run message like::
+
+      export JASPERINC="$EBROOTJASPER/include"
+ 
+The actual output depends on whether or not the required module for ``JasPer`` is available
+(see :ref:`extended_dry_run_overview_module_load_simulated_deps`).
+
+Silently defining environment variables
+#######################################
+
+The ``setvar`` function also supports defining environment variables *silently*, i.e. without producing a
+corresponding dry run message, via the named argument ``verbose``.
+
+This is used in a couple of places in the EasyBuild framework, to avoid some environment variables being defined
+cluttering the dry run output without added value. It can be used for similar reasons in easyblocks.
+
+For example, the ``PythonPackage`` uses it in the *install* step, to modify ``$PYTHONPATH`` as required by the
+``python setup.py install`` procedure (which is considered not relevant to include in the dry run output, since
+it's a technicality)::
+
+    env.setvar('PYTHONPATH', new_pythonpath, verbose=False)
+
+
+.. _extended_dry_run_guidelines_easyblocks_framework_functions_write_file:
+
+Writing or appending to files: ``write_file``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For writing and appending to files, the EasyBuild framework provides the ``write_file`` function (available from
+the ``easybuild.tools.filetools`` module).
+
+Using it is straightforward, for example::
+
+    write_file('example.txt', "Contents for the example file")
+
+To append to an existing file, ``write_file`` support a named argument ``append``.
+
+When used in dry run mode, ``write_file`` does not actually (attempt to) write to the file; instead, it just produces
+an appropriate dry run message and returns.
+
+For example::
+
+    file written: /tmp/eb-ksVC07/tmp.conf
+
+.. _extended_dry_run_guidelines_easyblocks_framework_functions_runtime_patching:
+
+Runtime patching of files: ``apply_regex_substitutions``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To make runtime patching of files in easyblocks easier, and to do it with taking the possibility of being in dry run
+module into account, the EasyBuild framework provides the ``apply_regex_substitutions`` function (available from the
```

Comment:
This should link to documentation about this feature.

Is this valid in an easyconfig?